cirq-google/cirq_google/named_topologies_test.py

```diff
@@ -0,0 +1,95 @@
+# Copyright 2021 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import itertools
+from typing import Iterable
+from unittest.mock import MagicMock
+
+import cirq
+import networkx as nx
+import pytest
+from cirq_google.named_topologies import (
+    draw_gridlike,
+    LineTopology,
+    DiagonalRectangleTopology,
+    get_placements,
+    plot_placements,
+)
+import cirq_google as cg
+
+
+def test_diagonal_rectangle_topology():
+    width = 2
+    height = 3
+    topo = DiagonalRectangleTopology(width, height)
+    assert all(1 <= topo.graph.degree[node] <= 4 for node in topo.graph.nodes)
+    assert topo.name == '2-3-diagonal-rectangle'
+    assert topo.n_nodes == topo.graph.number_of_nodes()
+
+    with pytest.raises(ValueError):
+        _ = DiagonalRectangleTopology(0, 3)
+    with pytest.raises(ValueError):
+        _ = DiagonalRectangleTopology(3, 0)
+
+    ax = MagicMock()
+    topo.draw(ax=ax)
+    ax.scatter.assert_called()
+
+    qubits = topo.nodes_as_gridqubits()
+    assert all(isinstance(q, cirq.GridQubit) for q in qubits)
+
+
+def test_line_topology():
+    n = 10
```

Comment:
It might be worth having a test with n=2 or something small and test the edges and n_nodes explicitly to make sure it is expected.

Reply:
added